auditlog_security/models/auditlog_rule.py

```diff
@@ -0,0 +1,118 @@
+            ("name", "not in", FIELDS_BLACKLIST),
+        ]
+
+    def unlink(self):
+        # if we delete auditlog rule, corresponding ir.rules are removed
+        # TODO PROPOSAL: a warning here with detailed information?
+        to_delete = self.get_linked_rules()
+        res = super(AuditlogRule, self).unlink()
+        if res:
+            res = res and to_delete.unlink()
+        return res
+
+    def add_default_group_if_needed(self):
+        self.ensure_one()
+        res = False
+        if not self.group_ids and self.field_ids:
+            # if group has been removed and no group specified ad base user default
+            res = self.with_context(no_iter=True).write(
+                {"group_ids": [(6, 0, [self.env.ref("base.group_user").id])]})
+        return res
+
+    @api.model
+    def create(self, vals):
+        res = super(AuditlogRule, self).create(vals)
+        res.add_default_group_if_needed()
+        if res.needs_rule():
+            res.generate_rules()
+        return res
+
+    @api.multi
+    def write(self, vals):
+        res = super(AuditlogRule, self).write(vals)
+        for this in self: 
+            added = this.add_default_group_if_needed()
+            if any([x in vals for x in ("group_ids", "field_ids", "model_id", "all_fields")]) or added:
+                if this.needs_rule():
+                    this.generate_rules()
+                else:
+                    this.get_linked_rules().unlink()
+        return res
+
+    def generate_rules(self):
+        old_rule = self.env["ir.rule"].search([("auditlog_id", "=", self.id)], limit=1)
+        values = self._prepare_rule_values()
+        if old_rule:
+            old_rule.with_context(auditlog_write=True).write(values)
+        else:
+            self.with_context(auditlog_write=True).env["ir.rule"].create(values)
+
+    def _prepare_rule_values(self):
+        domain_force = "[" + " ('log_id.model_id' , '=', %s)," % (self.model_id.id)
+        if self.field_ids:
+            domain_force += "('field_id', 'in',  %s)" % (self.field_ids.ids)
+        domain_force += "]"
+        return {
+            "name": "auditlog_extended_%s" % self.id,
+            "model_id": self.env.ref("auditlog.model_auditlog_log_line").id,
+            "groups": [(6, 0, self.group_ids.ids)],
+            "perm_read": True,
+            "domain_force": domain_force,
+            "auditlog_id": self.id,
+        }
+
+    @api.multi
+    def subscribe(self):
+        act_window_model = self.env['ir.actions.act_window']
+        for rule in self:
+            domain = "[('log_id.model_id', '=', %s), ('log_id.res_id', '=', active_id)]" % (
+                rule.model_id.id)
+            vals = {
+                'name': _("View log lines"),
+                'res_model': 'auditlog.log.line',
+                'src_model': rule.model_id.model,
+                'binding_model_id': rule.model_id.id,
+                'domain': domain,
```

Comment:
I think we should fill `groups_id` here with the result of `rule.mapped('groups_ids')`.

Then we should also make the `fields_ids` and `groups_ids` readonly once the rule is subscribed, to avoid us having to write functionality to keep the both in sync when groups are added or removed.